src/rulesDeploy.ts

```diff
@@ -175,15 +175,19 @@ export class RulesDeploy {
         if (answers.confirm) {
           // Find the oldest unreleased rulesets. The rulesets are sorted reverse-chronlogically.
           const releases: Release[] = await gcp.rules.listAllReleases(this.options.project);
-          const isReleasedFn = (ruleset: ListRulesetsEntry): boolean => {
-            return !!releases.find((release) => release.rulesetName === ruleset.name);
-          };
-          const unreleased: ListRulesetsEntry[] = _.reject(history, isReleasedFn);
+          const unreleased: ListRulesetsEntry[] = _.reject(
+            history,
+            (ruleset: ListRulesetsEntry): boolean => {
+              return !!releases.find((release) => release.rulesetName === ruleset.name);
+            }
+          );
           const entriesToDelete = unreleased.reverse().slice(0, RULESETS_TO_GC);
-          for (const entry of entriesToDelete) {
-            await gcp.rules.deleteRuleset(this.options.project, gcp.rules.getRulesetId(entry));
-            logger.debug(`[rules] Deleted ${entry.name}`);
-          }
+          await Promise.all(
+            entriesToDelete.map(async (entry) => {
+              await gcp.rules.deleteRuleset(this.options.project, gcp.rules.getRulesetId(entry));
+              logger.debug(`[rules] Deleted ${entry.name}`);
+            })
```

Comment:
please revert this change, these deletes should happen in series to avoid running into backend quota limits

((can you add a comment to that effect? this was a very reasonable optimization to make, we should explain why we didn't take it))

Reply:
Will do. Good eye.